tsinfer/inference.py

```diff
@@ -1174,7 +1174,14 @@ def __ancestor_find_path(self, ancestor, thread_index=0):
         end = ancestor.end
         assert ancestor.haplotype.shape[0] == (end - start)
         haplotype[start:end] = ancestor.haplotype
-        self._find_path(ancestor.id, haplotype, start, end, thread_index)
+        try:
+            self._find_path(ancestor.id, haplotype, start, end, thread_index)
```

Comment:
Rather than wrapping ``_find_path`` twice, perhaps we should put the try-except in the place that can fail in that function?

Reply:
Yep, but then we find it difficult to distinguish between ancestors and sample haplotypes, which is why I put it in twice.

It's actually a bit of a pain to find the ID of the sample haplotype (see my latest push) but worth it, I think.